Mods/src/PuppiMod.cc

```diff
@@ -141,235 +160,222 @@ void PuppiMod::SlaveBegin()
-void PuppiMod::SlaveTerminate()
+void
+PuppiMod::SlaveTerminate()
 {
   // ===== deallocate memory ====
-  // Or you know, don't
+  delete fPuppiParticles;
+  delete fMapper;
 }
 
 //--------------------------------------------------------------------------------------------------
-void PuppiMod::Process()
+void
+PuppiMod::Process()
 {
-  // Process entries of the tree. 
-  fVertexes = GetObject<VertexCol>(fVertexesName);
-  fPFCandidates = GetObject<PFCandidateCol>(fPFCandidatesName);
+  // Process entries of the tree.
+  auto* vertexes = GetObject<VertexCol>(fVertexesName);
+  if (!vertexes)
+    SendError(kAbortAnalysis, "Process", "PV collection not found");
+
+  auto* pv = vertexes->At(0);
 
-  PFCandidateArr *PuppiParticles = new PFCandidateArr;        // This array is only used if adding object to event, not publishing
+  auto* pfCandidates = GetObject<PFCandidateCol>(fPFCandidatesName);
 
   if (fDumpingPuppi) {
-    std::cout << "Primary Vertex location: " << fVertexes->At(0)->Position().x() << ", ";
-    std::cout << fVertexes->At(0)->Position().y() << ", " << fVertexes->At(0)->Position().z() << std::endl;
+    std::cout << "Primary Vertex location: " << pv->Position().x() << ", ";
+    std::cout << pv->Position().y() << ", " << pv->Position().z() << std::endl;
   }
 
   // This mapper will return particles that are only close in eta, phi space
-  ParticleMapper *Mapper = new ParticleMapper();
-  Mapper->Initialize(*fPFCandidates,fR0,fR0,fMaxEtas[fNumEtaBins-1]);
-
-  const Int_t numCandidates = fPFCandidates->GetEntries();
-  Double_t alphaF[numCandidates];                             // This is alpha(F) of all particles for particle i
-  Double_t alphaC[numCandidates];                             // This is alpha(C) of charged PV for particle i
-  Int_t IndicesF[numCandidates];                              // A bunch of indices used for sorting
-  Int_t IndicesC[numCandidates];
-  Int_t numCHPU[fNumEtaBins];                                 // Track the number of particles that are charged PU for eta bin
-  Int_t numFCHPUis0[fNumEtaBins];                             // Number of alphas to ignore when finding the median for each
-  Int_t numCCHPUis0[fNumEtaBins];                             //   eta bin
-  Double_t alphaFCHPU[fNumEtaBins][numCandidates];            // This is alphaF for charged PU particles for eta bin
-  Double_t alphaCCHPU[fNumEtaBins][numCandidates];            // This is alphaC for charged PU particles for eta bin
-  Int_t IndicesFCHPU[fNumEtaBins][numCandidates];
-  Int_t IndicesCCHPU[fNumEtaBins][numCandidates];
-
-  Int_t numCHPV[fNumEtaBins];                                 // Track the number of particles that are charged PV for eta bin
-  Double_t alphaFCHPV[fNumEtaBins][numCandidates];            // This is alphaF for charged PV particles for eta bin
-  Double_t alphaCCHPV[fNumEtaBins][numCandidates];            // This is alphaC for charged PV particles for eta bin
-
```

Comment:
See the general comments on vectors versus arrays. In this case, numCandidates is declared const but comes from an external variable whose value is indeterminate at compile time, and the construct above is possible only thanks to g++ extension of the C++ standard (I didn't even know this was possible). I think it internally converts the code to a new-delete, where delete happens when the array goes out of scope. Scram does use g++ to compile our code, but it's better to stick to standards.